sympy/vector/coordsysrect.py

```diff
@@ -170,6 +180,16 @@ def __str__(self, printer=None):
     def __iter__(self):
         return iter([self.i, self.j, self.k])
 
+    def __getattr__(self, name):
```

Comment:
`__getattr__` does not exactly 'overload' the dot operator I think. That's what `__getattribute__` does. `__getattr__ ` is only invoked if the attribute wasn't found the usual ways. If the attribute cannot be handled by `__getattr__` also, the usual `AttributeError` is raised. So effectively, there is no change with respect to the previous functioning of class except that before raising `AttributeError`, `__getattr__` is invoked to see if it can handle it.

In `sympy.diffgeom`, the `base_vector()` function with index must be called explicitly to access base vectors. However, it would be more convenient and logical to access them using the same `names` that the user has passed (which currently is only used for printing expressions). Issue #12492